tests/test_cost_tracking.py

```diff
@@ -0,0 +1,209 @@
+
+from .conftest import VCR_DEFAULT_MATCH_ON, CILLMModelNames
+
+
+@contextmanager
+def assert_costs_increased():
+    """All tests in this file should increase accumulated costs."""
+    initial_cost = GLOBAL_COST_TRACKER.lifetime_cost_usd
+    yield
+    assert GLOBAL_COST_TRACKER.lifetime_cost_usd > initial_cost
+
+
+class TestLiteLLMEmbeddingCosts:
+    @pytest.mark.asyncio
+    async def test_embed_documents(self):
+        stub_texts = ["test1", "test2"]
+        with assert_costs_increased():
+            async with cost_tracking_ctx():
+                model = LiteLLMEmbeddingModel(name="text-embedding-3-small", ndim=8)
+                await model.embed_documents(stub_texts)
+
+
+class TestLiteLLMModel:
+    @pytest.mark.vcr(match_on=[*VCR_DEFAULT_MATCH_ON, "body"])
+    @pytest.mark.parametrize(
+        "config",
+        [
+            pytest.param(
+                {
+                    "model_name": "gpt-4o-mini",
+                    "model_list": [
+                        {
+                            "model_name": "gpt-4o-mini",
+                            "litellm_params": {
+                                "model": "gpt-4o-mini",
+                                "temperature": 0,
+                                "max_tokens": 56,
+                            },
+                        }
+                    ],
+                },
+                id="chat-model",
+            ),
+            pytest.param(
+                {
+                    "model_name": "gpt-3.5-turbo-instruct",
+                    "model_list": [
+                        {
+                            "model_name": "gpt-3.5-turbo-instruct",
+                            "litellm_params": {
+                                "model": "gpt-3.5-turbo-instruct",
+                                "temperature": 0,
+                                "max_tokens": 56,
+                            },
+                        }
+                    ],
+                },
+                id="completion-model",
+            ),
+        ],
+    )
+    @pytest.mark.asyncio
+    async def test_call(self, config: dict[str, Any]) -> None:
+        with assert_costs_increased():
+            async with cost_tracking_ctx():
+                llm = LiteLLMModel(name=config["model_name"], config=config)
+                messages = [
+                    Message(role="system", content="Respond with single words."),
+                    Message(
+                        role="user", content="What is the meaning of the universe?"
+                    ),
+                ]
+                await llm.call(messages)
+
+    @pytest.mark.asyncio
+    async def test_call_w_figure(self) -> None:
+        async def ac(x) -> None:
+            pass
+
+        async with cost_tracking_ctx():
+            with assert_costs_increased():
+                llm = LiteLLMModel(name="gpt-4o")
+                image = np.zeros((32, 32, 3), dtype=np.uint8)
+                image[:] = [255, 0, 0]
```

Comment:
```suggestion
                image = np.full((32, 32, 3), [255, 0, 0], dtype=np.uint8)
```

Please feel free to ignore this too

Reply:
Yeah I didn't add this code - just mirroring what's in the other tests.